tools/run.py

```diff
@@ -0,0 +1,392 @@
+    nth_mixture['isFlippedMoving'] = np.append(nth_mixture['isFlippedMoving'], is_flipped_moving)
+    nth_mixture['speed'] = np.append(nth_mixture['speed'], ev_speed)
+    nth_mixture['rirs'].append(riridx)
+    return nth_mixture
+
+
+def populate_mixture(cfg, sample_counter, fold_event_durs, fold_event_filenames, n_traj, traj_doas, max_polyphony, nth_mixture):
+    event_counter = 0
+
+    # populate each "layer" in the mixture
+    for layer in range(max_polyphony):
+
+        event_idx_in_layer, event_counter, sample_counter = get_layer_event_indices(fold_event_durs, cfg.MIXTURE_DUR - cfg.MIXTURE_LAYER_SILENCE, event_counter, sample_counter)
+        nevents_in_layer = len(event_idx_in_layer)
+
+        silence_gaps = get_silence_gaps(nevents_in_layer, cfg.MIXTURE_LAYER_SILENCE, cfg.MIN_GAP_BTW_LAYER_EVENTS)
+
+        # distribute each event in the timeline
+        time_idx = 0
+        for nl in range(nevents_in_layer):
+
+            event_duration_nl, riridx, sample_onoffsets, ev_traj, is_moving, is_flipped_moving, ev_speed, event_nl, time_idx = get_event_riridx_onoffset(nl, silence_gaps, event_idx_in_layer, fold_event_durs, fold_event_filenames, cfg, time_idx, n_traj, traj_doas)
+
+            nth_mixture = update_nth_mixture(nth_mixture, riridx, time_idx, event_duration_nl, sample_onoffsets, traj_doas, nl, layer, ev_traj, fold_event_filenames, is_moving, is_flipped_moving, ev_speed, event_nl, cfg)
+
+            time_idx += event_duration_nl
+
+    sort_idx = np.argsort(nth_mixture['event_onoffsets'][:,0])
+    nth_mixture['files'] = nth_mixture['files'][sort_idx]
+    nth_mixture['class'] = nth_mixture['class'][sort_idx]
+    nth_mixture['event_onoffsets'] = nth_mixture['event_onoffsets'][sort_idx]
+    #nth_mixture['sample_onoffsets'] = nth_mixture['sample_onoffsets'][sort_idx]
+    nth_mixture['trajectory'] = nth_mixture['trajectory'][sort_idx]
+    nth_mixture['isMoving'] = nth_mixture['isMoving'][sort_idx]
+    nth_mixture['isFlippedMoving'] = nth_mixture['isFlippedMoving'][sort_idx]
+    nth_mixture['speed'] = nth_mixture['speed'][sort_idx]
+    nth_mixture['rirs'] = np.array(nth_mixture['rirs'],dtype=object)
+    nth_mixture['rirs'] = nth_mixture['rirs'][sort_idx]
+    new_doas = np.zeros(len(sort_idx),dtype=object)
+    new_sample_onoffsets = np.zeros(len(sort_idx),dtype=object)
+    upd_idx = 0
+    for idx in sort_idx:
+        new_doas[upd_idx] = nth_mixture['doa_azel'][idx].T
+        new_sample_onoffsets[upd_idx] = nth_mixture['sample_onoffsets'][idx]
+        upd_idx += 1
+    nth_mixture['doa_azel'] = new_doas
+    nth_mixture['sample_onoffsets'] = new_sample_onoffsets 
+
+    return nth_mixture
+
+def main():
+    '''
+    main function to trigger the data generation
+    '''
+
+    # parse config arguments
+    args = parse_args()
+    cfg = load_config(args, args.path_to_config)
+    print('generating data using parameters:')
+    print(yaml.dump(dict(cfg), allow_unicode=True, default_flow_style=False))
```

Comment:
Is there a better way to load and parse the `yaml` file? What about command-line arguments?

Reply:
Do we plan for command line arguments to overwrite existing `yaml` config settings? I think it might make sense to have a separate script for creating the config file to ensure the `yaml` file exists as an accurate log of the way data was generated.

Reply:
SGTM! Maybe having a log or metadata file output w/ the dataset so the generation parameters are known would be good, if we feel this is useful, I can open an issue and perhaps spend some time working on it.